### PR TITLE
Adapt CI to run UI tests on PR

### PR DESCRIPTION
**Pull request by tatsta**

### CODE REVIEW

File: .bin/upload-logs-and-notify.sh [hunk 1]
```diff
@@ -14,27 +14,41 @@ sleep ${RANDOM_VALUE}s
...
 
 export OPENCART_CURRENT_VERSION="${OPENCART_VERSION//-/.}"
 export PROJECT_FOLDER="opencart-ee-${OPENCART_CURRENT_VERSION}"
-GATEWAY_FOLDER=${REPO_NAME}/${PROJECT_FOLDER}/${GATEWAY}
-DATE_FOLDER=${GATEWAY_FOLDER}/${TODAY}
 
-if [ ! -d "${GATEWAY_FOLDER}" ]; then
-mkdir -p ${GATEWAY_FOLDER}
+if [ ! -d "${REPO_NAME}/${PROJECT_FOLDER}/${GATEWAY}" ]; then
+mkdir -p ${REPO_NAME}/${PROJECT_FOLDER}/${GATEWAY}
```
Comment (rinnhofer) [inline .bin/upload-logs-and-notify.sh]: missing indent

File: .bin/upload-logs-and-notify.sh [hunk 2]
```diff
@@ -14,27 +14,41 @@ sleep ${RANDOM_VALUE}s
...
 fi
 
-if [ ! -d "${DATE_FOLDER}" ]; then
-mkdir ${DATE_FOLDER}
+if [ ! -d "${REPO_NAME}/${PROJECT_FOLDER}/${GATEWAY}/${TODAY}" ]; then
+mkdir ${REPO_NAME}/${PROJECT_FOLDER}/${GATEWAY}/${TODAY}
```
Comment (rinnhofer) [inline .bin/upload-logs-and-notify.sh]: missing indent

File: .bin/run-ui-tests.sh [hunk 3]
```diff
@@ -2,4 +2,21 @@ set -a # automatically export all variables from .env file
...
+if [ ${TRAVIS_PULL_REQUEST} != "false" ]; then
+    export $GIT_BRANCH="${TRAVIS_PULL_REQUEST_BRANCH}"
+fi
+
+# find out test group to be run
+if [[ $GIT_BRANCH =~ "${PATCH_RELEASE}" ]]; then
+   TEST_GROUP="${PATCH_RELEASE}"
+elif [[ $GIT_BRANCH =~ "${MINOR_RELEASE}" ]]; then
+   TEST_GROUP="${MINOR_RELEASE}"
+# run all tests in nothing else specified
```
Comment (rinnhofer) [inline .bin/run-ui-tests.sh]: spelling mistake - still when we change it readability increases ;)

File: .bin/send-notify.sh [hunk 4]
```diff
@@ -29,8 +29,8 @@ curl -X POST -H 'Content-type: application/json' --data "{
         {
             'fallback': 'Failed test data',
             'text': 'There are failed tests.
-             Test report: ${PREVIEW_LINK}/${SCREENSHOT_COMMIT_HASH}/${PROJECT_FOLDER}/${GATEWAY}/${TODAY}/${REPORT_FILE} .
-             All screenshots can be found  ${REPO_LINK}/tree/${SCREENSHOT_COMMIT_HASH}/${PROJECT_FOLDER}/${GATEWAY}/${TODAY} .',
+             Test report: ${PREVIEW_LINK}/${SCREENSHOT_COMMIT_HASH}/${RELATIVE_REPORTS_LOCATION}/${REPORT_FILE} .
```
Comment (rinnhofer) [inline .bin/send-notify.sh]: We don't need single quotes here?

File: .travis.yml [hunk 5]
```diff
@@ -27,16 +44,23 @@ matrix:
...
+    #   or when build triggered ad-hoc with special commit message
+    #   tests will use extension version = branch which originated pull request
+    #   or branch which was defined in ad-hoc trigger in Travis UI
+    if: (type = pull_request AND (head_branch =~ env(PATCH_RELEASE) OR head_branch =~ env(MINOR_RELEASE) OR head_branch =~ env(MAJOR_RELEASE))) OR (commit_message =~ env(MANUAL_UITEST_TRIGGER_COMMIT))
     php: 7.1
     env:
     - GATEWAY=API-TEST
     - OPENCART_RELEASE_VERSION=1
     install:
     - export OPENCART_VERSION=$(awk "NR==$OPENCART_RELEASE_VERSION {print; exit}" ${OPENCART_RELEASES_FILE})
```
Comment (rinnhofer) [inline .travis.yml]: not sure, but I think this indent is incorrect - at least we had to change it within magento2 for the release

File: .bin/send-notify.sh (hunk 4 above)
Comment (tatsta) [reply .bin/send-notify.sh]: No, this is still part of `'text'`

File: .travis.yml (hunk 5 above)
Comment (tatsta) [reply .travis.yml]: I tested with our deploy script and it worked, but I still changed indentation for consistency

**Pull request merged**